From b4cab213246094aa575b0ca303b09fb213b1fd17 Mon Sep 17 00:00:00 2001
From: Alex Avance <aravance@gmail.com>
Date: Thu, 13 Jan 2011 17:46:46 -0600
Subject: [PATCH 06/11] Expose option to change the device hostname.

This adds an option to modify the device hostname used
in ip resolution. This is useful when connecting to the
android device in a dynamic dhcp environment.

Change-Id: I1d8302f06144dbb3e6ddba68594cc5718b4f0bb7
(based on commit Ibc145b74036617248d4f33c6866cc9c8a8cc8974)
---
 res/values/cm_strings.xml     |  3 +++
 res/xml/development_prefs.xml | 11 +++++++++++
 2 files changed, 14 insertions(+)

diff --git a/res/values/cm_strings.xml b/res/values/cm_strings.xml
index d1c58e46da..cd0759f42c 100644
--- a/res/values/cm_strings.xml
+++ b/res/values/cm_strings.xml
@@ -126,6 +126,9 @@
     <string name="security_settings_fingerprint_sensor_location_left">left side</string>
     <string name="security_settings_fingerprint_sensor_location_right">right side</string>
 
+    <!-- Hostname setting -->
+    <string name="device_hostname">Device hostname</string>
+
     <!-- Increasing ring tone volume -->
     <string name="increasing_ring_volume_option_title">Increasing ring volume</string>
     <string name="increasing_ring_min_volume_title">Start volume</string>
diff --git a/res/xml/development_prefs.xml b/res/xml/development_prefs.xml
index 87d109d94f..b58221bb6a 100644
--- a/res/xml/development_prefs.xml
+++ b/res/xml/development_prefs.xml
@@ -155,6 +155,17 @@
         <Preference android:key="clear_adb_keys"
                 android:title="@string/clear_adb_keys" />
 
+        <org.lineageos.internal.preference.HostnamePreference
+            android:key="device_hostname"
+            android:title="@string/device_hostname"
+            android:dialogTitle="@string/device_hostname"
+            android:positiveButtonText="@string/wifi_save"
+            android:negativeButtonText="@string/wifi_cancel"
+            android:selectAllOnFocus="true"
+            android:imeOptions="actionDone"
+            android:inputType="textNoSuggestions"
+            android:persistent="false" />
+
         <SwitchPreference
             android:key="enable_terminal"
             android:title="@string/enable_terminal_title"
-- 
2.17.1

